### DISCUSSION TITLE
Available on Mac ?

### DISCUSSION THREAD

**Discussion by trendland** [Comfy-Org/ComfyUI#63] [General]
Hi Guys. 
Comfy looks amazing and I'd love to test it out - Also Im on a Mac, (Sorry about that ;) 
Wondering if you know if it would be working so I can test your amazing new tool! 

Let me know and looking forward to play with it! 
Thank you in advance!

**Comment by 925-Studio** [comment]
Now it's seems only support cpu if you're using mac and obviously it's very slow. But it's an open source, so I believe MPS acceleration will be supported in the near future.

**Comment by moonpost** [comment]
Mac M1/M2 support, please.

**Reply by swankwc** [reply]
I second that.

**Reply by dileeparanawake** [reply]
M1 silicon / neural engine support would be amazing

**Reply by Black0S** [reply]
🖥️ I built a script that makes installing, launching & updating ComfyUI (and its dependencies) dead simple on **Apple Silicon**.

Feel free to try it out & share your feedback!
👉 https://github.com/Black0S/ComfyUI-Mac-Silicon-Launcher 🙌

**Comment by comfyanonymous** [comment]
I would like to support Mac and will try to implement MPS support but unfortunately I don't have a Mac or access to one.

**Reply by swankwc** [reply]
Just let us know when you have something worked up or need us to test something the community makes.

**Reply by HubertII** [reply]
Get one! Recently I've bought a mini with M2 - it's cheap and powerfull! My first Mac ever. Perhaps we could help you with collecting money?

**Comment by comfyanonymous** [comment]
Basic mac support has been implemented: https://github.com/comfyanonymous/ComfyUI/pull/244

**Reply by freshlesh3** [reply]
Thats great, are there instructions on how to install it on M1 Macs?

**Reply by MrSegundus** [reply]
Are there instructions on getting ComfyUI to work on Mac?

**Reply by Black0S** [reply]
> Are there instructions on getting ComfyUI to work on Mac?

🖥️ I built a script that makes installing, launching & updating ComfyUI (and its dependencies) dead simple on **Apple Silicon**.

Feel free to try it out & share your feedback!
👉 https://github.com/Black0S/ComfyUI-Mac-Silicon-Launcher 🙌

**Comment by junglehu** [comment]
MacOS intel chip only run on --cpu，  sigh～

**Comment by pontusi** [comment]
# Comfy on a M1/M2 Mac

This is how **I** got it to run. It would help if others tried as well and shared their experiences.

## Install Python 3

If you do not already have Python 3 installed, start by installing it. You can use this distribution

https://www.python.org/downloads/macos/

in which case you should remember to run the `Update Shell Profile.command` file after installation. I used the Python 3.10.11 version.

Open a new Terminal window and type

```
python3 --version
```

to verify that you have a properly installed Python 3.

## Download ComfyUI

If you have git installed then you can do a normal git clone to get the source. If you don't have git installed then you can download the zip-file instead.

I would suggest you put ComfyUI in `~/Applications/ComfyUI`, that is, the dedicated app directory in your home directory, but that is not required.

## Install dependencies

This is very similar to the Linux instructions except you have to explicitly use pip3.

Open a Terminal window in the ComfyUI directory and run

```
pip3 install -r requirements.txt
```

## Running Comfy

Open a Terminal window in the ComfyUI directory and run

```
python3 main.py
```

If it says "Setting vram state to: MPS" then you should be good. If not I can't help you.

**Reply by creynders** [reply]
Still works like a charm ...

**Reply by ryanbooker** [reply]
I'm seeing "Setting vram state to: SHARED", but "Device: maps"? I followed the instructions on the README.md, and have the nightly of PyTorch installed.

```
Total VRAM 65536 MB, total RAM 65536 MB
Set vram state to: SHARED
Device: mps
VAE dtype: torch.float32
```

And I get total garbage output unless I force cpu only rendering. e.g. with SDXL base 1.0 I just get a completely blue image for the default prompt/workflow. With cpu only I get the expected image, but obviously it's much slower.

**Reply by Black0S** [reply]
🖥️ I built a script that makes installing, launching & updating ComfyUI (and its dependencies) dead simple on **Apple Silicon**.

Feel free to try it out & share your feedback!
👉 https://github.com/Black0S/ComfyUI-Mac-Silicon-Launcher 🙌

**Comment by FreeBlues** [comment]
The current version can be run on my MacBookPro(Intel chip), with a eGPU(RX6800 16G).

**Reply by danch99** [reply]
is it fast ? I have a Mac mini intel chip, and I would like to transform it to a ComfyUI machine.
Nvidia GPUs are not supported by MacOS. Does it worth investing in a non Nvidia eGPU ?

**Comment by polarNerd204** [comment]
I used instruction from this thread and it works great….i am however having trouble understanding the GPU vs RAM usage. I’m not expecting 4090 speeds but with 128g of ram on a M2 Ultra, activity monitors are showing it only is using 3-5% cpu and upto 70% ….what is the rest of it doing or why is the rest of it not being used for faster results?. There must be some setting I’m unaware of.  If anyone knows please pass it along.

**Comment by buggz2k** [comment]
I have been using Easy Install for a long time now, works good.
https://github.com/Tavris1/ComfyUI-Easy-Install/tree/MAC-Linux
I just wish the nodes and workflows were not cuda tuned, sigh...

**Reply by Black0S** [reply]
🖥️ I built a script that makes installing, launching & updating ComfyUI (and its dependencies) dead simple on Apple Silicon.

Feel free to try it out & share your feedback!
👉 https://github.com/Black0S/ComfyUI-Mac-Silicon-Launcher 🙌
No nodes pre install ; only comfyui-manager It's all.

**Comment by vSnake87** [comment]
use https://www.comfy.org for apple silicon and you can run quantization on MPS see how:
#12378
